Clean/Circuit/Loops.lean

```diff
@@ -269,8 +281,8 @@ theorem mapM_forAll {xs : List α} :
 
 theorem mapM_vector_forAll {xs : Vector α n} :
   (xs.mapM circuit |>.operations m).forAll prop ↔
-    ∀ x ∈ xs, ∀ (i : ℕ) (_ : (x, i) ∈ xs.zipIdx), (circuit x |>.operations (m + i*lawful.local_length)).forAll prop := by
-  rw [mapM_forAll_vector_iff_list, mapM_forAll_iff_forM, ←Vector.forM_toList, forM_vector_forAll]
+    ∀ i : Fin n, (circuit xs[i.val] |>.operations (m + i*lawful.local_length)).forAll prop := by
+  rw [mapM_forAll_vector_iff_list, mapM_forAll_iff_forM, ←Vector.forM_toList, forM_vector_forAll']
```

Comment:
I changed the `mapM` statement to just provide the index and index into the input vector. I found this easier to use in a proof than the original